web/middlewares/instance.go

```diff
@@ -70,3 +70,18 @@ func SetInstance() gin.HandlerFunc {
 		c.Set("instance", instance)
 	}
 }
+
+// GetInstance will return the instance linked to the given gin context
+func GetInstance(c *gin.Context) (*Instance, error) {
```

Comment:
I would tend to agree. For commodity reasons and also because if we do not have an instance in the context, `panic`ing seems the right thing to do...